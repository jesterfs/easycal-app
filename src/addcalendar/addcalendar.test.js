import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import AddCalendar from './addcalendar'
import renderer from 'react-test-renderer';

describe('AddCalendar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <AddCalendar />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('renders the UI as expected', () => {
    const div = document.createElement('div')
    const tree = renderer
      .create(<BrowserRouter>
        <AddCalendar />
      </BrowserRouter>,
      div)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

});