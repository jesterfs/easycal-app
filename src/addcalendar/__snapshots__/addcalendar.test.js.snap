// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCalendar component renders the UI as expected 1`] = `
<div
  className="AddCalendarGreet, greetgroup"
>
  <div
    className="item"
  >
    <h2>
      Add a Calendar
    </h2>
  </div>
  <div
    className="item"
  >
    <form
      className="add-calendar-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="calendarName"
        >
          Calendar Name
        </label>
        <input
          id="calendarName"
          name="calendarName"
          placeholder="Calendar Name"
          type="text"
        />
      </div>
      <button
        className="addbtn"
        type="submit"
      >
        Create Calendar
      </button>
    </form>
  </div>
</div>
`;
